### PR TITLE
Damping sample

### PR DESCRIPTION
**Pull request by romainf-unity**
New damping sample, cute programmer cat, lots of comments in code.

InfoText displays this:
```
Damping Sample

In play mode,
- Select "ProgrammerCat" GameObject
- Move it around and rotate it
```

### CODE REVIEW

File: Assets/animation-jobs-samples/Samples/Scripts/Damping/Damping.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    void Update()
+    {
+        if (!m_Graph.IsValid())
+            return;
+
+        // Update job's delta time.
+        var dampingJob = m_DampingPlayable.GetJobData<DampingJob>();
+        dampingJob.deltaTime = Time.deltaTime;
```
Comment (mecanimdev) [inline Assets/animation-jobs-samples/Samples/Scripts/Damping/Damping.cs]: always use animationStream.deltaTime inside a job, if the user setup his animator to AnimatePhysics the deltaTime will be wrong

File: Assets/animation-jobs-samples/Runtime/AnimationJobs/DampingJob.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        rootHandle.SetRotation(stream, rootRotation);
+    }
+
+    /// <summary>
+    /// Procedurally generate the joints rotation.
+    /// </summary>
+    /// <param name="stream">The animation stream</param>
+    public void ProcessAnimation(AnimationStream stream)
+    {
+        if (jointHandles.Length < 2)
```
Comment (mecanimdev) [inline Assets/animation-jobs-samples/Runtime/AnimationJobs/DampingJob.cs:36]: not very important but validation should be made from the monobehaviour and throw an exception if the setup is not correct

File: Assets/animation-jobs-samples/Runtime/AnimationJobs/DampingJob.cs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        var parentRotation = rootRotation * localRotations[0];
+        positions[0] = parentPosition;
+        for (var i = 1; i < jointHandles.Length; ++i)
+        {
+            // The target position is the global position, without damping.
+            var newPosition = parentPosition + (parentRotation * localPositions[i]);
+
+            // Apply damping on this target.
+            var velocity = velocities[i];
+            newPosition = Vector3.SmoothDamp(positions[i], newPosition, ref velocity, 0.15f, Mathf.Infinity, deltaTime);
```
Comment (mecanimdev) [inline Assets/animation-jobs-samples/Runtime/AnimationJobs/DampingJob.cs]: use stream.deltaTime

**Review by mecanimdev — APPROVED** [review]

File: Assets/animation-jobs-samples/Runtime/AnimationJobs/DampingJob.cs (hunk 3 above)
Comment (romainf-unity) [reply Assets/animation-jobs-samples/Runtime/AnimationJobs/DampingJob.cs]: fixed!

File: Assets/animation-jobs-samples/Runtime/AnimationJobs/DampingJob.cs (hunk 2 above)
Comment (romainf-unity) [reply Assets/animation-jobs-samples/Runtime/AnimationJobs/DampingJob.cs:36]: discussed offline: checking it here allows to have a job that is easier use elsewhere.

**Review by davidg-unity — APPROVED** [review]

**Pull request merged**